lib/api/src/ptr.rs

```diff
@@ -60,7 +60,7 @@ pub type WasmPtr64<T> = WasmPtr<T, Memory64>;
 #[repr(transparent)]
 pub struct WasmPtr<T, M: MemorySize = Memory32> {
     offset: M::Offset,
-    _phantom: PhantomData<*mut T>,
```

Comment:
There used to be some obscure effects on type layout and optimizations when using full types, hence `PhantomData` is usually used with pointers rather than actual types.

Any specific reason you changed this?

Reply:
indeed that is the case